### PR TITLE
replace 50+ occurrences of print() with logger

### PR DESCRIPTION
**Pull request by richbeales**
### Background

We have a logging library implementation, we should use it - replace all relevant print() lines with an appropriate logger call (logger.debug/info/warn).  Needs manual testing to ensure i haven't broken anything

If anyone wants to collaborate, ping me - there's some circular dependencies with the autogpt.logs logger i'm trying to work through

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by ntindle** [comment]
Using `./run.sh` within a Codespace returns
```
  File "/workspaces/Auto-GPT/scripts/check_requirements.py", line 6, in <module>
    from autogpt.logs import logger
ModuleNotFoundError: No module named 'autogpt'
...
<Continues with install>
```

The rest seems to work though

**Review by arrenv — APPROVED** [review]
Looks good and no breaking changes on my end.

**Review by ntindle — APPROVED** [review]

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -83,11 +83,7 @@ def start_interaction_loop(self):
                 logger.typewriter_log(
                     "Continuous Limit Reached: ", Fore.YELLOW, f"{cfg.continuous_limit}"
                 )
-                send_chat_message_to_user(
-                    f"Continuous Limit Reached: \n {cfg.continuous_limit}"
```
Comment (Wladastic) [inline autogpt/agent/agent.py:87]: why was this removed?
